microcosm_caching/decorators.py

```diff
@@ -0,0 +1,112 @@
+from functools import wraps
+from logging import Logger
+from time import perf_counter
+from typing import Any, Type
+
+from marshmallow import Schema
+from microcosm.errors import NotBoundError
+from pymemcache.exceptions import MemcacheError
+
+from microcosm_caching.base import CacheBase
+
+
+DEFAULT_TTL = 60 * 60  # Cache for an hour by default
+
+
+def get_metrics(graph):
+    try:
+        return graph.metrics
+    except NotBoundError:
+        return None
+
+
+def cache_key(cache_prefix, key):
+    return f"{cache_prefix}:{key}"
+
+
+def cached(component, schema: Type[Schema], cache_prefix: str, ttl: int = DEFAULT_TTL):
+    """
+    Intended mainly for use as a decorator around microcosm_flask.CRUDStoreAdapters, namely
```

Comment:
Your comment is too self-deprecating. I might be explicit about how exactly this is special to the CRUDStoreAdapter and then explain what it does, so something like:

- caches the result of *decorated_fn* but there are some caveats that the result of *decorated_fn* must be json serializable. We make some assumptions about how to do this so this works for CRUDStoreAdapter methods
- requires that resource is identifiable by a single id

Reply:
Sure.